xpdacq/utils.py

```diff
@@ -92,39 +93,79 @@ def export_userScriptEtc():
 
-    Files can be compreesed or .yml, once imported, bt.list() should show updated acquire object list
+    Return
+    ------
+        moved_list : list
+        a list of file names that have been moved successfully
     '''
+    _f_ext_dst_dict = ['py', 'npy', 'yml']
     src_dir = glbl.import_dir
-    dst_dir = glbl.yaml_dir
     f_list = os.listdir(src_dir)
     if len(f_list) == 0:
-        print('INFO: There is no pre-defined user objects in {}'.format(src_dir))
+        print('INFO: There is no predefined user objects in {}'.format(src_dir))
         return 
-    # two possibilites: .yml or compressed files; shutil should handle all compressed cases
-    moved_f_list = []
+    # unpack every archived file in Import/
     for f in f_list:
-        full_path = os.path.join(src_dir, f)
-        (root, ext) = os.path.splitext(f)
-        if ext == '.yml':
-            shutil.copy(full_path, dst_dir)
-            moved_f_list.append(f)
-            # FIXME - do we want user confirmation?
-            os.remove(full_path)
-        else:
-            try:
-                shutil.unpack_archive(full_path, dst_dir)
-                moved_f_list.append(f)
-                # FIXME - do we want user confirmation?
-                os.remove(full_path)
-            except ReadError:
-                print('Unrecongnized file type {} is found inside {}'.format(f, src_dir))
+        try:
+            # shutil should handle all compressed cases
+            tar_full_path = os.path.join(src_dir, f)
+            shutil.unpack_archive(tar_full_path, src_dir)
+        except ReadError:
+            pass
+    f_list = os.listdir(src_dir) # new f_list, after unpack
+    moved_list = []
+    failure_list = []
+    for f_name in f_list:
+        if os.path.isfile(os.path.join(src_dir, f_name)):
+            src_full_path = os.path.join(src_dir, f_name)
+            (root, ext) = os.path.splitext(f_name)
+            if ext == '.yml':
+                dst_dir = glbl.yaml_dir
+                yml_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir)
+                moved_list.append(yml_dst_name)
+            elif ext == '.py':
+                dst_dir = glbl.usrScript_dir
+                py_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir) 
+                moved_list.append(py_dst_name)
+            elif ext == '.npy':
+                dst_dir = glbl.config_base
+                npy_dst_name = _copy_and_delete(f_name, src_full_path, dst_dir) 
+                moved_list.append(npy_dst_name)
+            elif ext in ('.tar', '.zip', '.gztar'):
                 pass
-    return moved_f_list
+            else:
+                print('{} is not a supported format'.format(f_name))
+                failure_list.append(f_name)
+                pass
+        else:
+            # don't expect user to have directory
+            print('Expect a file but get a directory {}. Did you properly archive it?'.format(f_name))
```

Comment:
replace with:
"I can only import files, not directories.  Please place in the import directory either (1) all your files such as scripts, masks and xpdAcq object yaml files or (2) a tar or zipped-tar archive file containing those files.